### PR TITLE
DATAJPA-1377 - Add JpaRepository.getById for consistency, deprecated JpaRepository.g…

### LINKED ISSUE #1697: JpaRepository.getOne(ID id) method name is inconsistent with other xById(ID id) method names [DATAJPA-1377]
**[Nathan Niesen](https://jira.spring.io/secure/ViewProfile.jspa?name=nniesen)** opened **[DATAJPA-1377](https://jira.spring.io/browse/DATAJPA-1377?redirect=false)** and commented

The other JpaRepository xOne(ID id) methods were changed to xById(ID id) but the getOne method was not

---

**Affects:** 2.0.8 (Kay SR8)

### PR DESCRIPTION
**Pull request by JesseW28**
…etOne.

Also added two new tests where getById is used instead of getOne.

Closes #1697

- [ ] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [ ] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

File: src/test/java/org/springframework/data/jpa/repository/AbstractPersistableIntegrationTests.java [hunk 1]
```diff
@@ -66,4 +68,16 @@ void equalsWorksForProxiedEntities() {
 
 		assertThat(proxy).isEqualTo(proxy);
 	}
+
+	@Test // DATAJPA-1377
```
Comment (SKrabbenborg) [inline src/test/java/org/springframework/data/jpa/repository/AbstractPersistableIntegrationTests.java:72]: These should note the gh issue number. Apparently the DATAJPA are the old Jira identifiers

File: src/test/java/org/springframework/data/jpa/repository/UserRepositoryTests.java [hunk 2]
```diff
@@ -987,6 +988,15 @@ void looksUpEntityReference() {
 		assertThat(result).isEqualTo(firstUser);
 	}
 
+	@Test // DATAJPA-1377
```
Comment (SKrabbenborg) [inline src/test/java/org/springframework/data/jpa/repository/UserRepositoryTests.java:991]: Ditto

**Review by SKrabbenborg — CHANGES_REQUESTED** [review]
Please check if your actual first and last name are used in your git config. Currently you have JesseW28 as username in `git log`. 

`git config --get user.name` should return Jesse Wouters. Can be set by running `git config --global "Jesse Wouters"`

**Comment by JesseW28** [comment]
@SKrabbenborg coming back on your comment: "Please check if your actual first and last name are used in your git config. Currently you have JesseW28 as username, see git log.

git config --get user.name should return Jesse Wouters. Can be set by running git config --global "Jesse Wouters"

The command git config --get user.name returns Jesse Wouters

**Pull request closed without merging**